File: frontend/src/types/candidateProfile.ts
export interface CandidateProfile {
  id: number;
  userId: number;
  fullName: string;
  email: string;
  phone: string;
  location: string;
  skills: string;
  education: string;
  experienceSummary: string;
  resumeUrl: string | null;
  updatedAt: string;
}

export interface UpdateCandidateProfileRequest {
  phone: string;
  location: string;
  skills: string;
  education: string;
  experienceSummary: string;
}

export interface UpdateCandidateProfileResponse {
  message: string;
  profile: CandidateProfile;
}

export interface ResumeUploadResponse {
  message: string;
  data: {
    fileName: string;
    downloadUrl: string;
    uploadedAt: string;
  };
}

File: backend/TalentSyncAI.Backend/TalentSyncAI.Api/Models/Entities/CandidateProfile.cs
namespace TalentSyncAI.Api.Models.Entities
{
    public class CandidateProfile
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public string Phone { get; set; } = string.Empty;

        public string Location { get; set; } = string.Empty;

        public string Skills { get; set; } = string.Empty;

        public string Education { get; set; } = string.Empty;

        public string ExperienceSummary { get; set; } = string.Empty;

        public string? ResumeUrl { get; set; }

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public User User { get; set; } = null!;
    }
}